cppcache/src/DistributedSystemImpl.cpp

```diff
@@ -143,11 +148,26 @@ void DistributedSystemImpl::unregisterCliCallback(int appdomainId) {
   }
 }
 
+std::string DistributedSystemImpl::getThreadName() {
+  std::thread::id id = std::this_thread::get_id();
+  std::stringstream ss;
+  ss << id;
+  std::string threadName;
+  if (g_threadNames[id] != "") {
```

Comment:
This is not a thread safe access to a non-thread safe data structure. There is actually no reason to use a map with thread ID as key since that is pretty much what `thread_local` is.

Reply:
Using thread_local now.